### PR TITLE
Site sync refactor

### PR DESCRIPTION
**Pull request by carlosmartinez**
## Context

This is a refactoring to facilitate the work of using Site UUIDs from TTAPI.

## Changes proposed in this pull request

Extract a class `AssignSiteAttributes` in order to reduce length and complexity of the `SyncSites` job.

## Link to Trello card

https://trello.com/c/pMwtRgL2/5069-site-uuids-refactor-syncsites

### CODE REVIEW

File: app/services/teacher_training_public_api/hydrate_site.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
+module TeacherTrainingPublicAPI
+  class HydrateSite
```
Comment (cwrw) [inline app/services/teacher_training_public_api/hydrate_site.rb:2]: could we call this `sync_site` to keep the naming consistent as we use `sync` everywhere else?

Comment (carlosmartinez) [reply app/services/teacher_training_public_api/hydrate_site.rb:2]: i did initially call it that, but decided on `hydrate` as the actual syncing is still happening back in `sync_sites` (as i didn't want to throw `@updates`, `@changeset` and `@incremental_sync` all round the codebase. but no major objection to calling it `SyncSite` if you feel it's a better fit.

Comment (cwrw) [reply app/services/teacher_training_public_api/hydrate_site.rb:2]: that's fair, I've tried thinking of a way we can surface changes from within this class, which allows us to `save!` the site here and just expose that method but tbh its not a good pattern to follow so happy to keep as is, maybe rename to `AssignSiteAttributes` but happy either way with this or `Hydrate`

Comment (cwrw) [inline app/services/teacher_training_public_api/hydrate_site.rb:2]: could you also add a test for this class please?

File: app/services/teacher_training_public_api/sync_sites.rb [hunk 2]
```diff
@@ -20,29 +20,7 @@ def perform(provider_id, recruitment_cycle_year, course_id, incremental_sync = t
         course_code: @course.code,
       ).includes(:location_status).paginate(per_page: 500)
 
-      sites.each do |site_from_api|
-        site = provider.sites.create_or_find_by(code: site_from_api.code) do |s|
-          # We need to set the name here so that the record is valid when created.
```
Comment (cwrw) [inline app/services/teacher_training_public_api/sync_sites.rb:25]: could we maintain this comment in the new class?

**Review by cwrw — COMMENTED** [review]
looks good glad we're breaking up all those loong methods, only minor comments. Also could you reword the commits from `wip` please?

File: spec/services/teacher_training_public_api/assign_site_attributes_spec.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
+require 'rails_helper'
+
+RSpec.describe TeacherTrainingPublicAPI::AssignSiteAttributes do
+  let(:site) { described_class.new(site_from_api, provider).call }
+
+  let(:provider) { create(:provider) }
+  let(:site_from_api) do
+    Hashie::Mash.new(
```
Comment (cwrw) [inline spec/services/teacher_training_public_api/assign_site_attributes_spec.rb:8]: I think using this these days isn't recommended and we aren't in this codebase, could we maybe use the TTAPI helpers or some form of struct instead, also 👍 for the test data, its time for an early dinner 🧆 !

**Review by cwrw — APPROVED** [review]
Thanks for addressing comments 👍

**Pull request merged**